src/OAuth2User.ts

```diff
@@ -23,6 +24,11 @@ export interface OAuth2UserOptions {
   token?: Token;
 }
 
+export type TokenListener = (tokens: Token) => void;
+export type TokenRefreshFailedListener = (error: Error) => void;
+export type EventName= "tokenRefresh" | "tokenRefreshFailed";
```

Comment:
Can you rename to `TokenEventName`?

Reply:
Or will we have other events?

Reply:
> good idea, this then should also should be used by the webln provider and exposed there [master/src/webln/OauthWeblnProvider.ts#L29-L38](https://github.com/getAlby/alby-js-sdk/blob/master/src/webln/OauthWeblnProvider.ts?rgh-link-date=2023-07-20T09%3A10%3A57Z#L29-L38)

how the webln provider will use it? I think keeping it general will be good